python/lsst/meas/base/wrappers.py

```diff
@@ -442,3 +444,201 @@ def _init(self, ctrl, name, mapper, logName=None):
+    GenericPluginClass : `type`, subclass of `GenericPlugin`
+        `GenericPlugin` subclass to wrap.
+    """
+    class SingleFrameFromGenericConfig(GenericPluginClass.ConfigClass, SingleFramePluginConfig):
+        pass
+
+    @register(name)
+    class SingleFrameFromGenericPlugin(SingleFramePlugin):
+        ConfigClass = SingleFrameFromGenericConfig
+
+        def __init__(self, config, name, schema, metadata, logName=None):
+            SingleFramePlugin.__init__(self, config, name, schema, metadata, logName=logName)
+            self._generic = GenericPluginClass(config, name, schema, metadata)
+
+        def measure(self, measRecord, exposure):
+            center = measRecord.getCentroid()
+            return self._generic.measure(measRecord, exposure, center)
+
+        def measureN(self, measCat, exposure, refCat, refWcs):
+            return self._generic.measureN(measCat, exposure, refCat, refWcs)
+
+        def fail(self, measRecord, error=None):
+            self._generic.fail(measRecord, error if error is not None else None)
+
+        @classmethod
+        def getExecutionOrder(cls):
+            return GenericPluginClass.getExecutionOrder()
+
+        def getTransformClass(self):
+            return self._generic.getTransformClass()
+
+    return SingleFrameFromGenericPlugin
+
+
+def forcedPluginFromGeneric(name, GenericPluginClass):
+    """Produce a ForcedPlugin sub-class from a GenericPlugin class
+
+    The class is also registered.
+
+    Parameters
+    ----------
+    name : `str`
+        Name of plugin to register.
+    GenericPluginClass : `type`, subclass of `GenericPlugin`
+        `GenericPlugin` subclass to wrap.
+    """
+    class ForcedFromGenericConfig(GenericPluginClass.ConfigClass, ForcedPluginConfig):
+        pass
+
+    @register(name)
+    class ForcedFromGenericPlugin(ForcedPlugin):
+        ConfigClass = ForcedFromGenericConfig
+
+        def __init__(self, config, name, schemaMapper, metadata, logName=None):
+            ForcedPlugin.__init__(self, config, name, schemaMapper, metadata, logName=logName)
+            schema = schemaMapper.editOutputSchema()
+            self._generic = GenericPluginClass(config, name, schema, metadata)
+
+        def measure(self, measRecord, exposure, refRecord, refWcs):
+            center = exposure.getWcs().skyToPixel(refWcs.pixelToSky(refRecord.getCentroid()))
+            return self._generic.measure(measRecord, exposure, center)
+
+        def measureN(self, measCat, exposure, refCat, refWcs):
+            return self._generic.measureN(measCat, exposure, refCat, refWcs)
+
+        def fail(self, measRecord, error=None):
+            self._generic.fail(measRecord, error if error is not None else None)
+
+        @classmethod
+        def getExecutionOrder(cls):
+            return GenericPluginClass.getExecutionOrder()
+
+        def getTransformClass(self):
+            return self._generic.getTransformClass()
+
+    return ForcedFromGenericPlugin
```

Comment:
I think I would prefer to see these as class methods on the generic plugin class called something like make<Single/forced>Plugin, so that something that is wrapped can just be singleFooPlugin = genericFooPluging.makeSingle("name"), each of these could then even deffer to a third more common code base to reduce duplication further if desired

Reply:
Sounds like a good idea. I'll give it a go.